paasta_tools/cli/cmds/logs.py

```diff
@@ -1230,6 +1239,48 @@ def print_logs_by_time(
         for line in aggregated_logs:
             print_log(line["raw_line"], levels, raw_mode, strip_headers)
 
+    def tail_logs(
+        self,
+        service: str,
+        levels: Sequence[str],
+        components: Iterable[str],
+        clusters: Sequence[str],
+        instances: List[str],
+        pods: Iterable[str] = None,
+        raw_mode: bool = False,
+        strip_headers: bool = False,
+    ) -> None:
+        stream_name = get_log_name_for_service(service, prefix="app_output")
+        endpoint = self.get_nats_endpoint_for_cluster(clusters[0])
+        if not endpoint:
+            raise NotImplementedError(
+                "Tailing logs is not supported in this cluster yet, sorry"
+            )
+
+        async def tail_logs_from_nats() -> None:
```

Comment:
this is a lot nicer than what the `scribereader` version of this does :p

Reply:
Thanks! Yeah the nats-py stuff is all async, so have to use it here, but it turned out quite clean which is great.

I'm not sure how this'll do concurrency-wise compared to the scribereader version, but I think that'll have to wait until prod to really see with something high-volume. From the NATS resources side of things at least it's been perfectly fine on load which is a good sign.